File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/context/events/async/SampleEventListener.kt
package io.micronaut.docs.context.events.async

// tag::imports[]
import io.micronaut.docs.context.events.SampleEvent
import io.micronaut.runtime.event.annotation.EventListener
import io.micronaut.scheduling.annotation.Async
import java.util.concurrent.atomic.AtomicInteger
// end::imports[]
import javax.inject.Singleton

// tag::class[]
@Singleton
open class SampleEventListener {

    var invocationCounter = AtomicInteger(0)

    @EventListener
    @Async
    open fun onSampleEvent(event: SampleEvent) {
        println("Incrementing invocation counter...")
        invocationCounter.getAndIncrement()
    }
}
// end::class[]

File: test-suite-python/src/test/python/micronaut/docs/context/events/application/SampleEventListener.py
# tag::imports[]
from jakarta.inject import Singleton
from micronaut.context.event import ApplicationEventListener
from micronaut.docs.context.events.SampleEvent import SampleEvent
# end::imports[]

# tag::class[]
@Singleton
class SampleEventListener(ApplicationEventListener[SampleEvent]):
    invocation_count: int = 0

    def onApplicationEvent(self, event: SampleEvent):
        self.invocation_count += 1
# end::class[]
